component/prometheus/exporter/cloudwatch/config.go

```diff
@@ -0,0 +1,219 @@
+	Auth       RegionAndRoles `river:",squash"`
+	CustomTags Tags           `river:"custom_tags,attr,optional"`
+	SearchTags Tags           `river:"search_tags,attr,optional"`
+	Type       string         `river:"type,attr"`
+	Metrics    []Metric       `river:"metric,block"`
+}
+
+// Tags represents a series of tags configured on an AWS resource. Each tag is a
+// key value pair in the dictionary.
+type Tags map[string]string
+
+// StaticJob will scrape metrics that match all defined dimensions.
+type StaticJob struct {
+	Name       string         `river:",label"`
+	Auth       RegionAndRoles `river:",squash"`
+	CustomTags Tags           `river:"custom_tags,attr,optional"`
+	Namespace  string         `river:"namespace,attr"`
+	Dimensions Dimensions     `river:"dimensions,attr"`
+	Metrics    []Metric       `river:"metric,block"`
+}
+
+// RegionAndRoles exposes for each supported job, the AWS regions and IAM roles in which the agent should perform the
+// scrape.
+type RegionAndRoles struct {
+	Regions []string `river:"regions,attr"`
+	Roles   []Role   `river:"role,block,optional"`
+}
+
+type Role struct {
+	RoleArn    string `river:"role_arn,attr"`
+	ExternalID string `river:"external_id,attr,optional"`
+}
+
+// Dimensions are the label values used to identify a unique metric stream in CloudWatch.
+// Each key value pair in the dictionary corresponds to a label value pair.
+type Dimensions map[string]string
+
+type Metric struct {
+	Name       string        `river:"name,attr"`
+	Statistics []string      `river:"statistics,attr"`
+	Period     time.Duration `river:"period,attr"`
+}
+
+// SetToDefault implements river.Defaulter.
+func (a *Arguments) SetToDefault() {
+	*a = defaults
+}
+
+// ConvertToYACE converts the river config into YACE config model. Note that the conversion is
+// not direct, some values have been opinionated to simplify the config model the agent exposes
+// for this integration.
+func ConvertToYACE(a Arguments) (yaceConf.ScrapeConf, error) {
+	var discoveryJobs []*yaceConf.Job
+	for _, job := range a.Discovery {
+		discoveryJobs = append(discoveryJobs, toYACEDiscoveryJob(job))
+	}
+	var staticJobs []*yaceConf.Static
+	for _, stat := range a.Static {
+		staticJobs = append(staticJobs, toYACEStaticJob(stat))
+	}
+	conf := yaceConf.ScrapeConf{
+		APIVersion: "v1alpha1",
+		StsRegion:  a.STSRegion,
+		Discovery: yaceConf.Discovery{
+			ExportedTagsOnMetrics: yaceModel.ExportedTagsOnMetrics(a.DiscoveryExportedTags),
+			Jobs:                  discoveryJobs,
+		},
+		Static: staticJobs,
+	}
+
+	// Run the exporter's config validation. Between other things, it will check that the service for which a discovery
+	// job is instantiated, it's supported.
+	if err := conf.Validate(); err != nil {
+		return yaceConf.ScrapeConf{}, err
+	}
+	patchYACEDefaults(&conf)
+
+	return conf, nil
+}
+
+// patchYACEDefaults overrides some default values YACE applies after validation.
+func patchYACEDefaults(yc *yaceConf.ScrapeConf) {
```

Comment:
is there any way that we can reuse the same functions that are in `pkg/integrations/cloudwatch_exporter/config.go`?

Reply:
Some yes some no. For example this `patchFunctions` can be used, but making type conversion re-usable will make the code much more complicated that just re-doing those. I'll do another pass trying to see what we can re-use.

Reply:
Maybe i'm missing something, but I thought what we are doing is convert from river to yaml (to configure the integration) and then from yaml to internal yace config?